### PR TITLE
add hapi integration

### PR DESCRIPTION
**Pull request by rochdev**
This PR adds support for the `hapi` framework. Since most of the logic was already available in the `express` integration, it was extracted so it can be shared between both frameworks.

### CODE REVIEW

**Comment by gbbr** [comment]
If you write documentation for each function, their arguments and how to use them from a public API standpoint I'm happy to review these for you.

**Comment by rochdev** [comment]
Added documentation, and also refactored to make it a bit easier to read. Do note however that this is currently not a public API (only for internal use in our integrations). I will add further documentation when the API becomes public.

File: src/plugins/util/web.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  },
+
+  // Start a span and activate a scope for a request.
+  instrument (tracer, config, req, res, name, callback) {
+    const childOf = tracer.extract(FORMAT_HTTP_HEADERS, req.headers)
+    const span = tracer.startSpan(name, { childOf })
+    const scope = tracer.scopeManager().activate(span)
+
+    if (config.service) {
+      span.setTag(SERVICE_NAME, config.service)
```
[the remark this answers is not in the document]
Comment (rochdev) [reply src/plugins/util/web.js]: The default service name is set by the tracer when starting any new span.

File: src/plugins/util/web.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+    callback && callback(span)
+
+    wrapEnd(req)
+
+    return span
+  },
+
+  // Reactivate the request scope in case it was changed by a middleware.
+  reactivate (req) {
```
[the remark this answers is not in the document]
Comment (rochdev) [reply src/plugins/util/web.js]: This is only used in integrations, where the scope could be changed by external factors in-between middleware.

**Pull request merged**